include/dr/shp/shp.hpp

```diff
@@ -9,8 +9,6 @@
 #include <dr/detail/ranges_shim.hpp>
 #include <dr/detail/segments_tools.hpp>
```

Comment:
This is a general comment and not specific to this PR. For mhp.hpp, my intention is to remove everything that is not a direct dependence of a user program, so I would get rid of `ranges_shim.hpp`, `iterator_adaptor.hpp`, ...

It should mostly be everything in algorithms/containers/views directories + implementation of the CPO's like dr::ranges::segments and local.

Reply:
Great! This is how mature libraries look like, e.g. [boost/filesystem.hpp](https://github.com/boostorg/filesystem/blob/fcc1bf3eca15323b90f2df042464668b0d131698/include/boost/filesystem.hpp) does not include any details. Details are included by particular headers that need them, e.g. [here in in boost/filesystem/directory.hpp](https://github.com/boostorg/filesystem/blob/fcc1bf3eca15323b90f2df042464668b0d131698/include/boost/filesystem/directory.hpp#L30).

@BenBrock, can we/I do the same in [shp.hpp](https://github.com/oneapi-src/distributed-ranges/blob/main/include/dr/shp/shp.hpp) that is 

- remove dr/detail/ from there 
- also move dr/shp/shp.hpp one level up to dr/shp.hpp in order to make it consistent with mhp, and includes style used e.g. by mentioned boost libs

?

Reply:
> * also move dr/shp/shp.hpp one level up to dr/shp.hpp in order to make it consistent with mhp, and includes style used e.g. by mentioned boost libs

And oneapi style agrees: https://github.com/oneapi-src/distributed-ranges/blob/main/doc/developer/design/oneapi-cpp-style-guide.md#additional-recommendations-on-includes